lib/types/result-stream.js

```diff
@@ -19,7 +21,32 @@ ResultStream.prototype._read = function() {
     this._readableState.reading = false;
   }
   while (!this.paused && this.buffer.length > 0) {
-    this.paused = this.push(this.buffer.shift());
+    this.paused = !this.push(this.buffer.shift());
+  }
+  if ( !this.paused && !this.buffer.length && this._readNext ) {
+    // console.log( "_read", this.paused, this.buffer.length );
+    this._readNext();
+    this._readNext = null;
+  }
+};
+
+/**
+ * Allows for throttling, helping nodejs keep the internal buffers reasonably sized.
+ * @param {function} function that triggers reading the next result chunk
```

Comment:
`@param` should follow the type and name of the parameter:
```
* @param {function} readNext function that triggers reading the next result chunk
```

Remove the `@constructor` jsdoc?